### PR TITLE
plugins/none-ls: add enableLspFormat option

### LINKED ISSUE #755: Lsp-format doesn't work easily with none-ls
To make none-ls work with lsp-format you need to add a custom onAttach:

```
none-ls.onAttach = "require('lsp-format').on_attach";
```

This could be automatic depending on if lsp-format is enabled, or an extra option on none-ls.

### PR DESCRIPTION
**Pull request by GaetanLepage**
This PR adds an option in `none-ls` to configure the use of `lsp-format` with it.

I also changed the behavior of `mkRaw` to ensure that `mkRaw ""` gives `null` instead of `""`.

Fixes #755

### CODE REVIEW

**Comment by GaetanLepage** [comment]
I don't know if this integration should be enabled automatically if the user has `lsp-format` enabled...

File: plugins/none-ls/default.nix [hunk 1]
```diff
@@ -28,6 +28,15 @@ in {
         description = "Plugin to use for none-ls";
       };
 
+      enableLspFormat = mkOption {
+        type = types.bool;
+        default = false;
```
Comment (traxys) [inline plugins/none-ls/default.nix:33]: we could maybe set the default to `plugins.lsp-format.enable` indeed?

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/none-ls/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/none-ls/default.nix:33]: Yes this makes sense.

Comment (GaetanLepage) [reply plugins/none-ls/default.nix]: Done.

**Review by traxys — APPROVED** [review]

**Pull request merged**